### DISCUSSION TITLE
dynamic particles

### DISCUSSION THREAD

**Discussion by idlebear** [TheCodez/dynamic-occupancy-grid-map#90] [Q&A]
I have the rviz plugin working for displaying the velocity outputs, but the values are a lot more dynamic than I'm expecting.  In Carla, sitting still there's a lot of colour showing up where nothing should be moving.  In gazebo with the jackal sim, again sitting still, the gridcells seem to have random velocities.  Here's a vid of the setup:

https://user-images.githubusercontent.com/11653808/177198631-23468af0-5eb6-400e-8160-fbfbeb89244d.mp4

And here are some of the mean_x/y_vel reported by getGridCells():

```
...
(9730) x: 1.814526, y: 0.403938
(9731) x: 1.967703, y: 0.046213
(9732) x: 1.528175, y: 0.096707
(9733) x: 1.160133, y: 0.093182
(9739) x: -0.763744, y: 0.295925
(9740) x: 0.901271, y: 0.546722
(9741) x: 0.822438, y: 0.667723
(9742) x: 0.554166, y: 0.099766
(9743) x: 0.988246, y: -0.005239
(9744) x: 1.471395, y: 0.002887
(9914) x: 0.075954, y: -0.886901
(9916) x: 0.787212, y: -0.900795
(9917) x: -1.374719, y: -0.858376
(9918) x: 0.500808, y: -0.300921
(9919) x: 0.120100, y: -0.490384
(9920) x: -0.272829, y: 0.503453
(9921) x: 0.127194, y: 0.903747
(9922) x: 1.422631, y: -2.129350
(9923) x: 1.060151, y: -0.105056
(9924) x: 1.115667, y: -0.347313
(9925) x: 2.396093, y: -0.929997
(9926) x: 0.609607, y: 1.076100
(9929) x: 1.747452, y: 0.510634
(9942) x: -0.366327, y: 0.143168
...
```

Am I interpreting this correctly?  For instance cell 9922 above appears to be moving at  just over 2m/s?

Thanks!

**Comment by TheCodez** [comment]
I would suggest decreasing `stddev_process_noise_position` and `stddev_process_noise_velocity` and see if that helps. Obviously static occupancy shouldn’t have such velocities. But I don’t think there’s necessarily anything wrong with the code. Often it‘s just the parameters.

**Comment by cbachhuber** [comment]
Agreed that there's nothing wrong with the code: this false positive movement is expected, Nuss also describes this in [his paper](https://arxiv.org/pdf/1605.02406.pdf), see Figure 14. He suggests combination with a radar to suppress these false positives. If you don't have a radar, you can only play with the parameters to reduce these false positives.

**Comment by idlebear** [comment]
Fair enough and not totally unexpected for a probabilistic system to have some amount of variability.  I was surprised to see how much I was seeing relative to the small spots indicated in the paper.  As @TheCodez  suggests, tuning the noise parameters reduces the variability, but can also suppress the ability to detect motion.  After tuning, I'll probably still need a filter of some sort to smooth everything out.

And ya, no radar...

**Reply by TheCodez** [reply]
You could try the approach of https://arxiv.org/abs/1812.08449 to get a more sophisticated separation of dynamic and static cells.
